src/fsharp/CheckComputationExpressions.fs

```diff
@@ -1304,8 +1304,8 @@ let TcComputationExpression cenv env (overallTy: OverallTy) tpenv (mWhole, inter
             // TODO: consider allowing translation to BindReturn
             Some(translatedCtxt (mkSynCall "Bind" mMatch [matchExpr; consumeExpr]))
 
-        | SynExpr.TryWith (_mTry, innerComp, _mTryToWith, _mWith, clauses, _mWithToLast, mTryToLast, spTry, _spWith) ->
-            let mTry = match spTry with DebugPointAtTry.Yes m -> m.NoteDebugPoint(RangeDebugPointKind.Try) | _ -> mTryToLast
+        | SynExpr.TryWith (innerComp, clauses, mTryToLast, spTry, _spWith, trivia) ->
+            let mTry = match spTry with DebugPointAtTry.Yes _ -> trivia.TryKeyword.NoteDebugPoint(RangeDebugPointKind.Try) | _ -> trivia.TryKeyword
```

Comment:
since the keyword ranges are on the new trivia field, this opens the door to removing the ranges from the `DebugPointAtTry.Yes` case in the future, since the data is the same.

Reply:
Good thinking, but the `try` keyword in `SynExpr.TryFinally` would also need to be captured first for this.

Reply:
This turned out to be a bit trickier after all. I'm going to leave that as is.